compiler/optimizer/OMRValuePropagation.hpp

```diff
@@ -281,118 +284,32 @@ class ValuePropagation : public TR::Optimization
    ValueConstraint *copyValueConstraints(ValueConstraints &valueConstraints);
 
 #if USE_TREES
-   class ValueConstraintHandler : public TREE_HANDLER<ValueConstraint>
+   class ValueConstraintHandler : public TREE_HANDLER <ValueConstraint>
       {
       public:
-      void setVP(TR::ValuePropagation *vp) {_vp = vp;}
```

Comment:
The critical issue here is that we are doing virtual inheritance (true virtual inheritance not extensible inheritance because VP is too complex to be complicated by the extensible class pattern) and we need to make sure that that the LocalValuePropagation and GlobalValuePropagation instances constructed in the OMROptimizationManager for the various optimization groups appropriately specialize when a language specific ValuePropagation has been declared. Daryl is not proposing a solution and we shouldn't pick a solution just because he asked a question.

The key thing we need to figure out is how the OMROptimizationManager can end up talking about the right optimizations specializations - having the OMR ValuePropagation classes talk about OMR::ValuePropagation and, for example, the J9 ValuePropagation classes talk about J9::ValuePropagation is fine. The external interface (where external interface is the set of methods called from outside these files) should be able to talk about TR::ValuePropagation and get the correct behavior. If we can figure that issue out then that is the design to go for.